src/overlays/actors/ovl_Door_Shutter/z_door_shutter.c

```diff
@@ -21,61 +47,689 @@ const ActorInit Door_Shutter_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_8099670C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80996840.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_809968D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80996A54.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80996B00.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80996B0C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80996C60.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80996D14.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80996E08.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80996EE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80996F98.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80997004.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80997150.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80997220.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_809973E8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80997528.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_80997568.s")
+typedef struct {
+    s16 objectId;
+    u8 index1;
+    u8 index2;
+} ShutterObjectInfo;
+
+static ShutterObjectInfo D_809980F0[] = {
+    { OBJECT_GND, 4, 4 },
+    { OBJECT_GOMA, 5, 5 },
+    { OBJECT_YDAN_OBJECTS, 0, 1 },
+    { OBJECT_DDAN_OBJECTS, 2, 2 },
+    { OBJECT_BDAN_OBJECTS, 3, 3 },
+    { OBJECT_GAMEPLAY_KEEP, 8, 8 },
+    { OBJECT_BDOOR, 7, 7 },
+    { OBJECT_GAMEPLAY_KEEP, 8, 8 },
+    { OBJECT_HIDAN_OBJECTS, 9, 10 },
+    { OBJECT_GANON_OBJECTS, 11, 11 },
+    { OBJECT_JYA_DOOR, 6, 6 },
+    { OBJECT_MIZU_OBJECTS, 12, 13 },
+    { OBJECT_HAKA_DOOR, 14, 15 },
+    { OBJECT_ICE_OBJECTS, 16, 16 },
+    { OBJECT_MENKURI_OBJECTS, 17, 17 },
+    { OBJECT_DEMO_KEKKAI, 18, 18 },
+    { OBJECT_OUKE_HAKA, 19, 19 },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Shutter/func_809975C0.s")
+typedef struct {
+    /* 0x0000 */ Gfx* a;
+    /* 0x0004 */ Gfx* b;
+    /* 0x0008 */ u8 c;
+    /* 0x0009 */ u8 translateZ;
+    /* 0x000A */ u8 e;
+    /* 0x000B */ u8 f;
+} ShutterInfo;
+
+static ShutterInfo D_80998134[] = {
+    { 0x060067A0, 0x0404B0D0, 130, 12, 20, 15 }, { 0x06006910, 0x0404B0D0, 130, 12, 20, 15 },
+    { 0x060000C0, 0x060001F0, 240, 14, 70, 15 }, { 0x06000590, 0x06006460, 0, 110, 50, 15 },
+    { 0x06012AB0, 0x00000000, 130, 12, 50, 15 }, { 0x0601EC20, 0x00000000, 130, 12, 50, 15 },
```

Comment:
I think I'll keep them 0 for now since really none of it makes sense as-is but it's just what we do until object decomp progresses further

Reply:
NULL is definitely more correct, but mixing "symbolic" (sure null isn't actually symbolic since it's just a define but I hope you get what I mean) and numeric values also doesn't seem very good at this stage. It seems more in-line with how we do things to handle everything at once in a later pass while adding object symbols, since no matter what I do to this now it won't be "correct" in the long term.

Reply:
i dont really follow. we definitely have multiple cases of using null for zeroed segment address fields. and even though we keep segments as hex numbers it doesnt change that they are pointers. They are going to be NULL both before and after we actually decompile the objects and implement names, so im not following what point there is in waiting

Reply:
We don't add the nonzero fields to undefined_syms and replace them with pointers so why should the zero fields be replaced with NULL? "Because it's a pointer" means we should also be doing the same for the others but we aren't, hence the thought to keep the zero values as numbers. It feels very inconsistent to do one and not the other, hence why I opt to wait until the codebase is in a position to do that. If we intend to do this consistently everywhere then I can by all means change it now, but I don't believe that's the case as it stands.

Reply:
null isnt a part of any object, so yeah theres no need to add it to undefined syms. youre just saying that its a pointer to nothing.
The segment addresses arent added to undefined syms because they are going to be replaced by their proper symbols when its time to do so, but NULL is NULL

Reply:
this is something we definitely do already, an example from player lib. and I think its more than fine to do so
```
Gfx* D_80125D68[] = {
    NULL, NULL, NULL, NULL, NULL, 0x06022688, NULL, 0x06022688,
};
```

Reply:
the main difference is that its pointless to make undefined syms entries for something when its uneeded and its gonna get scrapped anyway. 0 vs null is not in that same scenario. I understand your pov on the inconsistency, but its only temporary, and we arent losing anything by using the correct representation of 0 in this context when it costs 0 time to do correctly